projects/driver_controls_center_console/src/button_led.c

```diff
@@ -0,0 +1,116 @@
+    [EE_LIGHT_STATE_OFF] = CENTER_CONSOLE_EVENT_BUTTON_SET_STATE_OFF,
+  };
+
+  // If it's one of the Light statuses we care about then process it
+  switch (light_id) {
+    case EE_LIGHT_TYPE_LOW_BEAMS:
+      event_raise(state_to_local_event_map[light_state], EE_CENTER_CONSOLE_DIGITAL_INPUT_LOW_BEAM);
+      break;
+    case EE_LIGHT_TYPE_DRL:
+      event_raise(state_to_local_event_map[light_state], EE_CENTER_CONSOLE_DIGITAL_INPUT_DRL);
+      break;
+    case EE_LIGHT_TYPE_SIGNAL_HAZARD:
+      event_raise(state_to_local_event_map[light_state], EE_CENTER_CONSOLE_DIGITAL_INPUT_HAZARDS);
+      break;
+    // Currently we don't process the strobe LED in here, since we use the BPS
+    // heartbeat message instead to raise the event triggering a transition in
+    // the LED FSM.
+    default:
+      return STATUS_CODE_OK;
+  }
+
+  return STATUS_CODE_OK;
+}
+
+StatusCode button_led_init(GpioExpanderStorage *storage, ButtonLedGpioExpanderPins *pins) {
+  button_led_fsm_init();
+
+  // Create FSMs for each LED output
+  button_led_fsm_create(&s_button_led_fsms[CENTER_CONSOLE_BUTTON_LED_BPS], storage,
+                        EE_CENTER_CONSOLE_DIGITAL_INPUT_BPS, pins->bps_indicator, "BPSIndicator");
+  button_led_fsm_create(&s_button_led_fsms[CENTER_CONSOLE_BUTTON_LED_PWR], storage,
+                        EE_CENTER_CONSOLE_DIGITAL_INPUT_POWER, pins->power_indicator,
+                        "PowerIndicator");
+  // These are separate because Lights sends an explicit ON/OFF for these,
+  // unlike Power State
+  button_led_fsm_create(&s_button_led_fsms[CENTER_CONSOLE_BUTTON_LED_DRL], storage,
+                        EE_CENTER_CONSOLE_DIGITAL_INPUT_DRL, pins->lights_drl, "DRLIndicator");
+  button_led_fsm_create(&s_button_led_fsms[CENTER_CONSOLE_BUTTON_LED_LOW_BEAMS], storage,
+                        EE_CENTER_CONSOLE_DIGITAL_INPUT_LOW_BEAM, pins->lights_low_beams,
+                        "LowBeams");
+  button_led_fsm_create(&s_button_led_fsms[CENTER_CONSOLE_BUTTON_LED_HAZARDS], storage,
+                        EE_CENTER_CONSOLE_DIGITAL_INPUT_HAZARDS, pins->lights_hazards, "Hazards");
+
+  // Enable RX handler to update power state LED
+  status_ok_or_return(
+      can_register_rx_handler(SYSTEM_CAN_MESSAGE_POWER_STATE, prv_power_state_rx_handler, NULL));
+
+  // Enable RX handler to update car Lights LEDs
+  status_ok_or_return(
+      can_register_rx_handler(SYSTEM_CAN_MESSAGE_LIGHTS_STATE, prv_lights_state_rx_handler, NULL));
+
+  return STATUS_CODE_OK;
+}
+
+bool button_led_process_event(const Event *e) {
+  bool processed = false;
+  for (size_t i = 0; i < NUM_CENTER_CONSOLE_BUTTON_LEDS; ++i) {
+    // These are handled in the Radio FSM
+    if (i == CENTER_CONSOLE_BUTTON_LED_REVERSE || i == CENTER_CONSOLE_BUTTON_LED_NEUTRAL ||
+        i == CENTER_CONSOLE_BUTTON_LED_DRIVE) {
+      continue;
+    }
+    processed |= fsm_process_event(&s_button_led_fsms[i], e);
+  }
+  return processed;
```

Comment:
Is this boolean value used anywhere?

Reply:
Nope, you're absolutely right that it's not used anywhere.

The reason why this is here is to keep a consistent API with other places where we do the same thing. Consider a few examples:

* `libraries/ms-common/can`
* `projects/chaos/relay`
* `projects/plutus/ltc_afe`